File: R/Vigenere_German.r
source('PreProcess_German.r')
source('ASCII_30_Add.r')
source('ASCII_30_Del.r')
source('ASCII_Enc_German.r')
source('ASCII_Dec_German.r')
source('EncryptVigenere_German.r')
source('DecryptVigenere_German.r')

file <- read.delim("nachtlied-utf-8.txt")
rawtext <- paste(unlist(file), collapse = "")

plaintext <- PreProcess_German(rawtext)
plaintext_ascii <- utf8ToInt(plaintext)
plaintext_ascii_30 <- ASCII_30_Add(plaintext_ascii)

ciphertext_ascii_30 <-
  EncryptVigenere_German(plaintext_ascii_30, "bratwurst")
ciphertext_ascii <- ASCII_Enc_German(ciphertext_ascii_30)
ciphertext_ascii_dec <- ASCII_Dec_German(ciphertext_ascii_30)

dectext_ascii_30 <-
  DecryptVigenere_German(ciphertext_ascii_dec, "bratwurst")
dectext_ascii <- ASCII_30_Del(dectext_ascii_30)

ciphertext <- intToUtf8(ciphertext_ascii)
dectext <- intToUtf8(dectext_ascii)

sprintf("%s", substring(plaintext, 300, 400))
sprintf("%s", substring(ciphertext, 300, 400))
sprintf("%s", substring(dectext, 300, 400))

File: JavaScript/Vigenere_German.js
// Import required functions
import { PreProcess_German } from "./PreProcess_German.mjs";
import { ASCII_30_Add } from "./ASCII_30_Add.mjs";
import { EncryptVigenere_German } from "./EncryptVigenere_German.mjs";
import { ASCII_Enc_German } from "./ASCII_Enc_German.mjs";
import { ASCII_Dec_German } from "./ASCII_Dec_German.mjs";
import { DecryptVigenere_German } from "./DecryptVigenere_German.mjs";
import { ASCII_30_Del } from "./ASCII_30_Del.mjs";

// Process Plaintext
var rawtext = fs.readFileSync("nachtlied-utf-8.txt", "utf-8");
var plaintext = PreProcess_German(rawtext);
var plaintext_ascii = plaintext.split('').map(function (char) { return char.charCodeAt(0); });
var plaintext_ascii_30 = plaintext_ascii.map(ASCII_30_Add);

// Encrypt Plaintext
var ciphertext_ascii_30 = EncryptVigenere_German(plaintext_ascii_30);
var ciphertext_ascii = ciphertext_ascii_30.map(ASCII_Enc_German);
var ciphertext = String.fromCharCode.apply(null, ciphertext_ascii);

// Decrypt Ciphertext
ciphertext_ascii_30 = ciphertext_ascii.map(ASCII_Dec_German);
var dectext_ascii_30 = DecryptVigenere_German(ciphertext_ascii_30);
var dectext_ascii = dectext_ascii_30.map(ASCII_30_Del);
var dectext = String.fromCharCode.apply(null, dectext_ascii);

// Decrypt Ciphertext
console.log(plaintext.substring(300, 400));
console.log(ciphertext.substring(300, 400));
console.log(dectext.substring(300, 400));
